### PR TITLE
[PB-4458] Add Metrics inspect Support

### PR DESCRIPTION
**Pull request by aks-px**
**What this PR does / why we need it**:
For having metrics inspect support in px-backup repository which will be used for functional automation.

**Which issue(s) this PR fixes** (optional)
Closes #PB-4458

**Special notes for your reviewer**:
Have ran basic backup test creation and verified the InspectMetrics is working fine.Attaching below the link of successful run in jenkins and aetos :

https://jenkins.pwx.dev.purestorage.com/blue/organizations/jenkins/portworx-backup%2Fsystem-tests%2Fbyoc%2Fpx-backup-on-demand-system-test-byoc/detail/px-backup-on-demand-system-test-byoc/2675/pipeline

https://aetos.pwx.purestorage.com/resultSet/testSetID/351843

### CODE REVIEW

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: drivers/backup/backup.go [hunk 1]
```diff
@@ -435,6 +437,13 @@ type ActivityTimeLine interface {
 	EnumerateActivityTimeLine(ctx context.Context, req *api.ActivityEnumerateRequest) (*api.ActivityEnumerateResponse, error)
 }
 
+// Metrics object interface
+type Metrics interface {
+
```
Comment (mkoppal-px) [inline drivers/backup/backup.go:442]: nit: Remove this empty line here.

Comment (aks-px) [reply drivers/backup/backup.go:442]: Okay

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by ak-px — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Pull request merged**